### PR TITLE
Time since oracle update

### PR DESCRIPTION
**Pull request by burtrico**
Added a tooltip to the futures market price that shows the time since the last oracle update. 

## Description
First I added a tooltip which I made custom in order to cleanly add a timer function to the tooltip component rather than pass down a timer component prop. I also added a query component under queries, useRateUpdateQuery.ts, which queries Synthetix subgraph on optimism-main for the latest "rateUpdate" (oracle price update) which includes a timestamp. Using this query data inside of MarketDetails, I pass the latest oracle price update time to a "TimeTooltip" component that displays the time since the last oracle update using an imported "CountUpTimer" component. 

Currently the timer component only shows the updated time when the tooltip is opened/closed. I can change this to be a live counting timer if desired. 

I also added a styled component hover transform of 1.02 to the market asset perp price. This is very light, but gives the user a slight indication of where the tooltip source is.

## Related issue
[](https://github.com/Kwenta/kwenta/issues/618)

## Motivation and Context
Optimism Oracle price updates are slow. This allows users to have an idea when the last price update came in and then estimate when the next one will be (20min or however long).

## How Has This Been Tested?
I used brave browser, optimism mainnet, and metamask. My changes did not affect other components.

## Screenshots (if appropriate):
Screen Shot 2022-04-14 at 9 56 15 PM

### CODE REVIEW

File: components/Tooltip/TimerTooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	startTimeDate: Date | undefined;
+	children?: React.ReactNode;
+	preset?: string;
+	top?: string;
+	bottom?: string;
+	left?: string;
+	right?: string;
+}
+
+const GeneralTooltip = (props: ToolTipProps) => {
```
Comment (LeifuChen) [inline components/Tooltip/TimerTooltip.tsx:15]: Just curious about why you named this component as `GeneralTooltip` and then put `TimerTooltip` into it.

File: sections/futures/CountUpTimer/CountUpTimer.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		const interval = setInterval(() => setTime(calcTime()), 1000);
+
+		return () => {
+			clearInterval(interval);
+		};
+	}, [calcTime]);
+
+	return (
+		<Container>
+			<p>{`Time since oracle update: ${time.minutes}:${time.seconds} min`}</p>
```
Comment (LeifuChen) [inline sections/futures/CountUpTimer/CountUpTimer.tsx:43]: 1. Please move this text into `translation\en.json`. 
2. Is the current format by design? Can we show the last update like below in tooltip?

**Last Update**
18 min ago

**Review by LeifuChen — CHANGES_REQUESTED** [review]
@burtrico Good job. Just some small questions about your implementation. Thanks.

**Review by LeifuChen — CHANGES_REQUESTED** [review]
@burtrico Great job! Would you kindly change it to be a liver timer?

File: components/Tooltip/TimerTooltip.tsx [hunk 3]
```diff
@@ -38,45 +37,46 @@ const TimerTooltip = (props: ToolTipProps) => {
...
 		[calcTime, activeMouse, currentStartTime]
-	)
+	);
 
 	useEffect(() => {
-		if(newUpdate) { 
-			console.log("!! newUpdate =", newUpdate)
-			setNewUpdate(false)
+		if (newUpdate) {
+			console.log('!! newUpdate =', newUpdate);
```
Comment (LeifuChen) [inline components/Tooltip/TimerTooltip.tsx:74]: Same. Please remove the `console.log`.

File: components/Tooltip/TimerTooltip.tsx [hunk 4]
```diff
@@ -38,45 +37,46 @@ const TimerTooltip = (props: ToolTipProps) => {
...
 
-	useInterval(() => { 
-		if(activeMouse || !newUpdate) {
-			setTotalSeconds(calcTime()) 
-			console.log("useInterval =", totalSeconds)
+	useInterval(
+		() => {
+			if (activeMouse || !newUpdate) {
+				setTotalSeconds(calcTime());
+				console.log('useInterval =', totalSeconds);
```
Comment (LeifuChen) [inline components/Tooltip/TimerTooltip.tsx:65]: Please remove the `console.log` before pushing the codes. Thanks.

File: sections/futures/CountUpTimer/CountUpTimer.tsx [hunk 5]
```diff
@@ -43,13 +43,13 @@ export default function CountUpTimer({ startTimeDate }: Props) {
 		// if (currentStartTime && (currentStartTime !== startTimeDate))
 		// clearInterval(interval)
 		const interval = setInterval(() => {
-			setTotalSeconds(calcTime())
-			console.log("activeMouse =", activeMouse)
-			if(!activeMouse) clearInterval(interval)
+			setTotalSeconds(calcTime());
+			console.log('activeMouse =', activeMouse);
```
Comment (LeifuChen) [inline sections/futures/CountUpTimer/CountUpTimer.tsx:47]: Please remove the `console.log`.

**Review by LeifuChen — CHANGES_REQUESTED** [review]
1. Please clean the code following the lint error.
2. When we move the mouse on both the value of Price and Open Interest, the number is growing bigger. Is this animation by any design?
3. The timer always starts from `23 seconds go` when the prices updates. @Tburm is this from the subgraph?

**Comment by burtrico** [comment]
> 1. Please clean the code following the lint warnings.
> 2. When we move the mouse on both the value of Price and Open Interest, the number is growing bigger. Is this animation by any design?
> 3. The timer always starts from `23 seconds go` when the prices updates. @Tburm is this from the subgraph?



> 1. Lint errors are fixed now
> 2. Yes, I added a slight transform to the two values that have tooltips associated with them. Yashar approved (see below). My thinking is that highlighting/transforming the tooltip source values will clearly show the user what the tooltip is based on and where on the page.
> 3. I'm not sure about the 23 seconds, I'll let you know if I find anything out related to that. If it is consistent then that does seem odd.
Screen Shot 2022-04-28 at 10 20 06 AM

**Review by LeifuChen — APPROVED** [review]
@burtrico Good job! Looks good to me.

**Comment by LeifuChen** [comment]
@platschi The implementation looks fine. However, the rpc seems not very stable right now. So the oracle timestamp might not be updated in some cases (force refresh will help).

**Pull request merged**